<launch>
    <arg name="serial_name" default="/dev/ttyUSB0" />
    <arg name="baudrate" default="921600" />
    <arg name="send_buffer" default="100" />
    <arg name="send_freq" default="100" />
    <arg name="output" default="screen" />
    <arg name="self_id" default="4" />
    <node pkg="inf_uwb_ros" name="uwb_node" type="inf_uwb_ros_node" output="$(arg output)" >
        <param name="serial_name" value="$(arg serial_name)" />
        <param name="baudrate" value="$(arg baudrate)" type="int" />
        <param name="send_buffer" value="$(arg send_buffer)" type="int" />
        <param name="send_freq" value="$(arg send_freq)" type="double" />
        <param name="recv_freq" value="100" type="double" />
        <param name="self_id" value="$(arg self_id)" type="int"/>
        <remap from="/planning/swarm_traj_recv" to="/planning/swarm_traj" />
        <param name="groundnode" value="true" />

        <remap from="/swarm_expl/drone_state_send" to="/swarm_expl/drone_state_send_$(arg self_id)" />
        <remap from="/swarm_expl/drone_state_recv" to="/swarm_expl/drone_state_recv_$(arg self_id)" />
        <remap from="/swarm_expl/pair_opt_send" to="/swarm_expl/pair_opt_send_$(arg self_id)" />
        <remap from="/swarm_expl/pair_opt_recv" to="/swarm_expl/pair_opt_recv_$(arg self_id)" />
        <remap from="/swarm_expl/pair_opt_res_send" to="/swarm_expl/pair_opt_res_send_$(arg self_id)" />
        <remap from="/swarm_expl/pair_opt_res_recv" to="/swarm_expl/pair_opt_res_recv_$(arg self_id)" />
        <remap from="/swarm_expl/grid_tour_send" to="/swarm_expl/grid_tour_send_$(arg self_id)" />
        <remap from="/swarm_expl/hgrid_send" to="/swarm_expl/hgrid_send_$(arg self_id)" />
        <remap from="/multi_map_manager/chunk_stamps_send" to="/multi_map_manager/chunk_stamps_send_$(arg self_id)" />
        <remap from="/multi_map_manager/chunk_data_send" to="/multi_map_manager/chunk_data_send_$(arg self_id)" />
        <remap from="/multi_map_manager/chunk_stamps_recv" to="/multi_map_manager/chunk_stamps_recv_$(arg self_id)" />
        <remap from="/multi_map_manager/chunk_data_recv" to="/multi_map_manager/chunk_data_recv_$(arg self_id)" />
        <remap from="/planning/swarm_traj_recv" to="/planning/swarm_traj_recv_$(arg self_id)" />
        <remap from="/planning/swarm_traj_send" to="/planning/swarm_traj_send_$(arg self_id)" />
    </node>

</launch>
